tests/util_test.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+"""PyTest for :mod:`pykern.util`
+
+:copyright: Copyright (c) 2022 Bivio Software, Inc.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+import pytest
+
+
+def test_is_pure_text():
+    from pykern import pkunit
+    from pykern import util
+
+    pkunit.pkeq(True, util.is_pure_text(b""))
```

Comment:
test `b"\0"`, which will decode properly, but is incorrect. Here's how perl does it:

> The -T and -B tests work as follows. The first block or so of the file is examined to see if it is valid UTF-8 that includes non-ASCII characters. If so, it's a -T file. Otherwise, that same portion of the file is examined for odd characters such as strange control codes or characters with the high bit set. If more than a third of the characters are strange, it's a -B file; otherwise it's a -T file. Also, any file containing a zero byte in the examined portion is considered a binary file. (If executed within the scope of a [use locale](https://perldoc.perl.org/perllocale) which includes LC_CTYPE, odd characters are anything that isn't a printable nor space in the current locale.) If -T or -B is used on a filehandle, the current IO buffer is examined rather than the first block. Both -T and -B return true on an empty file, or a file at EOF when testing a filehandle. Because you have to read a file to do the -T test, on most occasions you want to use a -f against the file first, as in next unless -f $file && -T $file.

Reply:
I added the "\x00" heuristic check. Would there be value in adding back in the " If more than a third of the characters are strange" heuristic as well?

Reply:
You have to test for more characters than "\x00". Read the perl description again, and figure out what those might be.

Reply:
Also the check has to happen after the utf8 check. utf8 allows for "\x0" so check the result in try after the decode.